### PR TITLE
Support monitoring in windows

### PR DESCRIPTION
**Pull request by aiwantaozi**
Problem:
can't get windows container/node metrics

Solution:
add windows endpoint and service monitor

Issue:
https://github.com/rancher/rancher/issues/20076

### CODE REVIEW

**Comment by orangedeng** [comment]
Why do you add the changes to monitoring 0.0.3? It should be in 0.0.4.

**Comment by aiwantaozi** [comment]
updated, window wmi node exporter needs to be released before this chart gets merged

**Comment by orangedeng** [comment]
LGTM

**Comment by gitlawr** [comment]
Could you do a rebase? Quite a few conflicts now.

**Comment by aiwantaozi** [comment]
@gitlawr updated, thanks

File: charts/rancher-monitoring/v0.0.4/charts/exporter-node-windows/templates/service.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+kind: Service
+metadata:
+  name:   expose-node-metrics-windows
+  labels:
+    app: {{ template "app.name" . }}
+    chart: {{ template "app.version" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    os: windows
+    {{ .Values.apiGroup }}: "true"
```
Comment (gitlawr) [inline charts/rancher-monitoring/v0.0.4/charts/exporter-node-windows/templates/service.yaml:10]: Why do we need this kind of label? It is not that common in community charts I've seen.

File: charts/rancher-monitoring/v0.0.4/charts/exporter-node-windows/templates/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  resources:
+  - tokenreviews
+  verbs:
+  - create
+- apiGroups:
+  - authorization.k8s.io
+  resources:
+  - subjectaccessreviews
+  verbs:
+  - creat
```
Comment (gitlawr) [inline charts/rancher-monitoring/v0.0.4/charts/exporter-node-windows/templates/rbac.yaml]: Looks like a typo, is it valid? How are these RBAC resources used by a node-exporter?

**Comment by aiwantaozi** [comment]
@gitlawr  Updated, thanks

File: charts/rancher-monitoring/v0.0.4/charts/exporter-node-windows/templates/service.yaml (hunk 1 above)
Comment (aiwantaozi) [reply charts/rancher-monitoring/v0.0.4/charts/exporter-node-windows/templates/service.yaml:10]: removed

**Comment by gitlawr** [comment]
LGTM

File: charts/rancher-monitoring/v0.0.4/values.yaml [hunk 3]
```diff
@@ -164,6 +164,21 @@ exporter-node:
...
 
+exporter-node-windows:
+  enabled: false
+  apiGroup: "monitoring.coreos.com"
+  endpoints: []
+  nodeSelectors: []
+  image:
+    repository: micheliac/wmi-exporter-package
+    tag: windows-1809
+  ports:
```
Comment (loganhz) [inline charts/rancher-monitoring/v0.0.4/values.yaml:175]: tag should be something like `v0.0.1`

**Comment by loganhz** [comment]
Link to
 https://github.com/rancher/rancher/issues/20122
https://github.com/rancher/rancher/issues/20513
https://github.com/rancher/rancher/issues/20077

**Comment by cjellick** [comment]
@gitlawr and @orangedeng 's reviews were long ago. Can you please re-review?

**Comment by loganhz** [comment]
Sure

File: charts/rancher-monitoring/v0.0.4/values.yaml (hunk 3 above)
Comment (loganhz) [reply charts/rancher-monitoring/v0.0.4/values.yaml:175]: Please change repository to `rancher/wmi_exporter-package` and tag `v0.0.1`

https://hub.docker.com/r/rancher/wmi_exporter-package/tags

**Review by orangedeng — APPROVED** [review]

**Comment by aiwantaozi** [comment]
update image since `rancher/wmi_exporter-package:v0.0.1` is released

**Comment by orangedeng** [comment]
LGTM

**Review by cjellick — APPROVED** [review]
lgtm

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**